### PR TITLE
Fix removing locks in recursive mode

### PR DESCRIPTION
**Pull request by johaven**
Surely an oversight, in the recursive mode the child locks must be deleted. The "recursive" parameter must therefore be taken into account in the `get_url_lock_list` function

### CODE REVIEW

File: wsgidav/dav_provider.py [hunk 1]
```diff
@@ -850,9 +850,9 @@ def is_locked(self):
             return False
         return self.provider.lock_manager.is_url_locked(self.get_ref_url())
 
-    def remove_all_locks(self, *, recursive):
+    def remove_all_locks(self, *, recursive=False):
```
Comment (mar10) [inline wsgidav/dav_provider.py:853]: I'd prefer, not to introduce a default. The current idea is to force the caller to make an explicit statement.

File: wsgidav/lock_man/lock_manager.py [hunk 2]
```diff
@@ -278,14 +278,14 @@ def is_token_locked_by_user(self, token, principal):
         """Return True, if <token> exists, is valid, and bound to <principal>."""
         return self.get_lock(token, key="principal") == principal
 
-    def get_url_lock_list(self, url):
+    def get_url_lock_list(self, url, recursive=False):
```
Comment (mar10) [inline wsgidav/lock_man/lock_manager.py:281]: Let's force a named arg:
`def get_url_lock_list(self, url, *, recursive=False)`

File: wsgidav/lock_man/lock_manager.py [hunk 3]
```diff
@@ -324,10 +324,10 @@ def is_url_locked_by_token(self, url, lock_token):
         lockUrl = self.get_lock(lock_token, key="root")
         return lockUrl and util.is_equal_or_child_uri(lockUrl, url)
 
-    def remove_all_locks_from_url(self, url):
+    def remove_all_locks_from_url(self, url, recursive=False):
```
Comment (mar10) [inline wsgidav/lock_man/lock_manager.py:327]: Let's force a named arg:
`def remove_all_locks_from_url(self, url, *, recursive=False)`

**Review by mar10 — CHANGES_REQUESTED** [review]
Please add a line in the chagelog as well.

Thanks for contributing!

**Comment by johaven** [comment]
I think it's good, i have another PR to submit

File: wsgidav/fs_dav_provider.py [hunk 4]
```diff
@@ -104,8 +104,8 @@ def delete(self):
         if self.provider.readonly:
             raise DAVError(HTTP_FORBIDDEN)
         os.unlink(self._file_path)
-        self.remove_all_properties(recursive=True)
-        self.remove_all_locks(recursive=True)
+        self.remove_all_properties(recursive=False)
```
Comment (mar10) [inline wsgidav/fs_dav_provider.py:107]: Why was this behavior changed (from the docstring, the delete() method is supposed to be recursive)?

File: wsgidav/fs_dav_provider.py [hunk 5]
```diff
@@ -104,8 +104,8 @@ def delete(self):
...
+        self.remove_all_locks(recursive=False)
```
Comment (mar10) [inline wsgidav/fs_dav_provider.py:108]: see above

File: CHANGELOG.md [hunk 6]
```diff
@@ -31,8 +33,7 @@
 
 **Other changes**
 
-- Provider root paths are evaluated relative to the location of the configuration 
-  file 
+- Provider root paths are evaluated relative to the location of the configuration file
```
Comment (mar10) [inline CHANGELOG.md:36]: I did not see a typo here, do I miss something?
Generally text files should have a break at ~ 80 columns, although I admit that this is not always the case yet ;-)

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,8 @@
 
 ## 4.0.2 / Unreleased
 
+- #251 Fix removing locks in recursive mode
```
Comment (mar10) [inline CHANGELOG.md:5]: Thanks

File: wsgidav/fs_dav_provider.py (hunk 4 above)
Comment (johaven) [reply wsgidav/fs_dav_provider.py:107]: If it's a resource why should locks be removed recursively? For a collection it seems logical, we can find locks below the directories, but not for a file. Right ?

File: CHANGELOG.md (hunk 6 above)
Comment (johaven) [reply CHANGELOG.md:36]: Sorry I'm a fan of wide screens :p It's a long debate but my editor is configured on columns of 120 (PEP8 is a bit too restrictive for me in 2022 :p) I'll fix this line in next commit :)

File: wsgidav/fs_dav_provider.py (hunk 4 above)
Comment (johaven) [reply wsgidav/fs_dav_provider.py]: I understand, it's ok :)

**Pull request merged**